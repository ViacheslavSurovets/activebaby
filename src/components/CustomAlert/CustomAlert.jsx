import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectShowAlert } from '@redux/alert/alert.selectors';
import PropTypes from 'prop-types';
import { Portal } from '@components/Portal';
import { Alert } from './internal';


const CustomAlert = ( { showAlert } ) => (
  <>
    {
      showAlert.length > 0 && <Portal component={ Alert } />
    }
  </>
);


const mapStateToProps = createStructuredSelector ( {
  showAlert: selectShowAlert
} );

export default connect ( mapStateToProps ) ( CustomAlert );

CustomAlert.propTypes = {
  showAlert: PropTypes.array
};
