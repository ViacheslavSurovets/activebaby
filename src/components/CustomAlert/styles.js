import styled from 'styled-components';
import {mobile} from '@media';

export const AlertContainer = styled.section`
  display: flex;
  justify-content: space-around;
  align-items: center;
  z-index: 10;
  width: 40rem;
  height: 12rem;
  position:absolute;
  top: 10%;
  left: 50%;
  margin-left: -15rem ;
  background: ${ ( { theme } ) => theme.ghostWhite };
  ${mobile}{
    width: 95%;
    margin-left: -48.5%;
  };
`;

export const AlertText = styled.div`
  text-align: center;
`;
