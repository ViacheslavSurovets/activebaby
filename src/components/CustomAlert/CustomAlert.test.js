import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import CustomAlert from './CustomAlert';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('CustomAlert', ()=> {
  it('CustomAlert Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <CustomAlert  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
