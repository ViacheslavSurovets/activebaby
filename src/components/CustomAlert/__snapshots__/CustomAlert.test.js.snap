// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomAlert CustomAlert Snapshot 1`] = `ShallowWrapper {}`;
